### PR TITLE
Refactor polkadot-parachain service for more code reuse

### PR DESCRIPTION
**Pull request by skunert**
This is a refactoring, no changes to the logic are included (if you find some, report :D).

## Change Overview
In https://github.com/paritytech/polkadot-sdk/issues/2455, dependency on actual runtimes was removed for the system parachains. This means that the trait bounds we have on various `start_node_xy` do not check anything anymore, since they all point to the same runtime. Exception is asset-hub-polkadot which uses a different key type.

This PR unifies the different nodes as much as possible. `start_node_impl` is doing the heavy lifting and has been made a bit more flexible to support the rpc extension instantiation that was there before.

The generics for `Runtime` and `AuraId` have been removed where possible. The fake runtime is imported as `FakeRuntime` to make it very clear to readers that it is not the generic parameter.

Multiple nodes where using the same import queue/start_consensus closure, they have been unified.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5390975

**Review by michalkucharczyk — APPROVED** [review]

**Comment by michalkucharczyk** [comment]
I guess you wanted to merge origin/master?

**Comment by skunert** [comment]
> @skunert if you can merge master I can finish my review, currently there are lots of unrelated changes in the diff

@seadanda  Yes, sorry! Messed this somehow, fill fix asap.

File: cumulus/polkadot-parachain/src/service.rs [hunk 1]
```diff
@@ -2193,6 +1080,185 @@ pub async fn start_contracts_rococo_node(
 	.await
 }
 
+/// Start relay-chain consensus that is free for all.
+/// Everyone can submit a block, the relay-chain decides
+/// what is backed and included.
```
Comment (seadanda) [inline cumulus/polkadot-parachain/src/service.rs:1085]:
```suggestion
/// Start relay-chain consensus that is free for all. Everyone can submit a block, the relay-chain
/// decides what is backed and included.
```

File: cumulus/polkadot-parachain/src/service.rs [hunk 2]
```diff
@@ -1190,464 +578,56 @@ where
...
-#[sc_tracing::logging::prefix_logs_with("Parachain")]
-pub async fn start_asset_hub_lookahead_node<RuntimeApi, AuraId: AppCrypto + Send + Codec + Sync>(
-	parachain_config: Configuration,
-	polkadot_config: Configuration,
-	collator_options: CollatorOptions,
-	para_id: ParaId,
-	hwbench: Option<sc_sysinfo::HwBench>,
-) -> sc_service::error::Result<(TaskManager, Arc<ParachainClient<RuntimeApi>>)>
+/// Build the import queue for parachain runtimes
+/// that started with relay chain consensus and switched to aura.
```
Comment (seadanda) [inline cumulus/polkadot-parachain/src/service.rs:623]:
```suggestion
/// Build the import queue for parachain runtimes that started with relay chain consensus and
/// switched to aura.
```

File: cumulus/polkadot-parachain/src/service.rs [hunk 3]
```diff
@@ -2193,6 +1080,185 @@ pub async fn start_contracts_rococo_node(
...
+		para_id,
+		|_, _, _, _| Ok(RpcModule::new(())),
+		build_relay_to_aura_import_queue::<_, AuraId>,
+		start_lookahead_aura_consensus,
+		hwbench,
+	)
+	.await
+}
+
+/// Start a parachain node.
```
Comment (seadanda) [inline cumulus/polkadot-parachain/src/service.rs:1240]:
```suggestion
/// Start a parachain node for Rococo Contracts.
```

File: cumulus/polkadot-parachain/src/service.rs [hunk 4]
```diff
@@ -36,12 +36,13 @@ use cumulus_primitives_core::{
...
 };
 use cumulus_relay_chain_interface::{OverseerHandle, RelayChainInterface};
+use sc_rpc::DenyUnsafe;
 use sp_core::Pair;
 
 use jsonrpsee::RpcModule;
 
-use crate::{fake_runtime_api::aura::RuntimeApi, rpc};
-pub use parachains_common::{AccountId, Balance, Block, Hash, Header, Nonce};
+use crate::{fake_runtime_api::aura::RuntimeApi as FakeRuntimeApi, rpc};
```
Comment (seadanda) [inline cumulus/polkadot-parachain/src/service.rs:44]: Good idea, it was pretty confusing before.

**Review by seadanda — APPROVED** [review]
LGTM, nice refactor

**Pull request merged**